web/src/compose_ui.ts

```diff
@@ -17,36 +17,55 @@ import * as stream_data from "./stream_data";
 import * as user_status from "./user_status";
 import * as util from "./util";
 
+// TODO: Refactor to push this into a field of ComposeTriggeredOptions.
+type messageType = "stream" | "private";
+type ComposeTriggeredOptions = {
+    trigger: string;
+    private_message_recipient: string;
+    topic: string;
+    stream_id: number;
+};
```

Comment:
This type gets confusingly used for two unrelated things, also fixed in the next commit.